ports/libsimpleservo/jniapi/build.rs

```diff
@@ -3,31 +3,20 @@
  * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
 
 use std::env;
-use std::path::Path;
+use std::fs::File;
+use std::io::Write;
 
 fn main() {
-    // Get the NDK path from NDK_HOME env.
-    let ndk_path =
-        env::var_os("ANDROID_NDK").expect("Please set the ANDROID_NDK environment variable");
-    let ndk_path = Path::new(&ndk_path);
-
-    // compiling android_native_app_glue.c
-    let c_file = ndk_path
-        .join("sources")
-        .join("android")
-        .join("native_app_glue")
-        .join("android_native_app_glue.c");
-    cc::Build::new()
-        .file(c_file)
-        .warnings(false)
-        .compile("android_native_app_glue");
-
     // Get the output directory.
     let out_dir =
         env::var("OUT_DIR").expect("Cargo should have set the OUT_DIR environment variable");
 
-    println!("cargo:rustc-link-lib=static=android_native_app_glue");
+    // FIXME: we need this workaround since jemalloc-sys still links
+    // to libgcc instead of libunwind, but Android NDK 23c and above
+    // don't have libgcc. Alternatively, we could disable jemalloc
+    // for android in servo_allocator
+    let mut libgcc = File::create(out_dir.clone() + "/libgcc.a").unwrap();
+    libgcc.write_all(b"INPUT(-lunwind)").unwrap();
+
```

Comment:
I think it would simplify the build to just disable jemalloc on Android.

Reply:
The comment is outdated. I did try disabling jemalloc but that requires providing an implementation of [this](https://github.com/servo/servo/blob/main/components/allocator/lib.rs#L39) method. I couldn't find a way to do that with the standard allocator.

Reply:
Could you use `malloc_usable_size`? https://docs.rs/libc/latest/libc/fn.malloc_usable_size.html

Reply:
I did look into that as well, but the documentation [warns](https://doc.rust-lang.org/std/alloc/struct.System.html) against directly using the backend allocator methods, but I can switch to `malloc_usable_size` we think it should be safe in this specific case.

Reply:
Yeah, let's just use the backend allocator methods and disable jemaloc. I think it's fine in this case.